src/routes/join/components/SignupPage.js

```diff
@@ -1,93 +1,51 @@
-const { default: Input } = require('@material-ui/core/Input');
-const { default: InputAdornment } = require('@material-ui/core/InputAdornment');
-const { default: IconButton } = require('@material-ui/core/IconButton');
-const { default: Visibility } = require('@material-ui/icons/Visibility');
-const { default: VisibilityOff } = require('@material-ui/icons/VisibilityOff');
-const StrangeForms = require('strange-forms');
+const { default: FormControl } = require('@mui/material/FormControl');
+const { default: InputLabel } = require('@mui/material/InputLabel');
+const { default: Input } = require('@mui/material/Input');
+const { default: InputAdornment } = require('@mui/material/InputAdornment');
+const { default: IconButton } = require('@mui/material/IconButton');
+const { default: Visibility } = require('@mui/icons-material/Visibility');
+const { default: VisibilityOff } = require('@mui/icons-material/VisibilityOff');
 
 const internals = {};
 
-module.exports = class SignupPage extends StrangeForms(React.Component) {
+module.exports = function SignupPage({ reqCreateAccount }) {
 
-    static propTypes = {
-        reqCreateAccount: T.func.isRequired
-    };
+    const handleClickShowPassword = (ev) => {
 
-    static fields = {
-        firstName: '',
-        lastName: '',
-        email: '',
-        password: ''
+        setShowPassword(!showPassword);
     };
 
-    constructor(props) {
-
-        super(props);
-
-        this.state = {
-            firstName: '',
-            lastName: '',
-            email: '',
-            password: '',
-            showPassword: false,
-            isSubmitting: false
-        };
-
-        this.strangeForm({
-            fields: Object.keys(SignupPage.fields),
-            get: (someProps, field) => {
-
-                return someProps.data ? someProps.data[field] : '';
-            },
-            act: () => null
-        });
-    }
-
-    handleClickShowPassword = (ev) => {
-
-        this.setState((currentState) => ({
-            showPassword: !currentState.showPassword
-        }));
-    }
-
-    handleMouseDownPassword = (ev) => {
+    const handleMouseDownPassword = (ev) => {
 
         ev.preventDefault();
-    }
-
-    disableSubmit() {
+    };
 
-        const firstName = this.fieldValue('firstName');
-        const lastName = this.fieldValue('lastName');
-        const email = this.fieldValue('email');
-        const password = this.fieldValue('password');
+    const disableSubmit = () => {
 
         return !firstName || !lastName || !email || !password;
-    }
+    };
 
-    handleSubmit = async (ev) => {
+    const handleSubmit = async (ev) => {
 
         ev.preventDefault();
-        const accountInfo = this.formatFields();
-        this.setState({ isSubmitting: true });
+        const accountInfo = formatFields();
+        setIsSubmitting(true);
         const [err] = await this.props.reqCreateAccount(accountInfo);
```

Comment:
Might want to test this happy path out— I think this would currently throw an error.
```suggestion
        const [err] = await reqCreateAccount(accountInfo);
```

Reply:
Mhm! That's [also addressed here](https://github.com/BigRoomStudios/strangeluv/pull/273/files#diff-19c2c7a4a6b350795c07620ec34a8b7e2d6777c107b7fab30fc98977f1b37c85L41-L42).